src/components/EmojiList/EmojiList.jsx

```diff
@@ -0,0 +1,127 @@
+import React, { Fragment } from 'react';
+import PropTypes from 'prop-types';
+import styles from './EmojiList.scss';
+import classnames from 'classnames/bind';
+
+const cx = classnames.bind( styles );
+
+const getBulletStyle = ( bullet, id, i ) => {
+	if ( ! bullet ) {
+		return null;
+	}
+
+	return `.${id} li:nth-of-type(${i + 1})::before { content: '${bullet}'; }`;
+};
+
+const ListStyles = ( { bullets, id } ) => (
+	<style
+		dangerouslySetInnerHTML={{
+			__html: bullets
+				.map( ( bullet, index ) => getBulletStyle( bullet, id, index ) )
+				.join( '' ),
+		}}
+	/>
+);
+
+ListStyles.propTypes = {
+	bullets: PropTypes.array.isRequired,
+	id: PropTypes.string.isRequired,
+};
+
+const ListItems = ( {
+	id,
+	innerHtml,
+	tagName,
+} ) => (
+	React.createElement(
+		tagName,
+		{
+			className: `${styles.container} ${id}`,
+			dangerouslySetInnerHTML: { __html: innerHtml },
+		}
+	)
+);
+
+ListItems.propTypes = {
+	id: PropTypes.string.isRequired,
+	innerHtml: PropTypes.string.isRequired,
+	tagName: PropTypes.string.isRequired,
+};
+
+// If the emojis array contains emojis, strip those emojis from the supplied HTML
+const stripHtmlEmojis = ( html, emojis ) => {
+	if ( ! emojis.length ) {
+		return html;
+	}
+
+	const pattern = new RegExp( `[${emojis.join( '' )}]`, 'g' );
+	return html.replace( pattern, '' );
+};
+
+/**
+ * Displays a WP content block.
+ */
+const EmojiList = ( {
+	bullets,
+	id,
+	innerHtml,
+	tagName,
+} ) => (
+	<Fragment>
+		<ListStyles bullets={bullets} id={id} />
+		<ListItems
+			id={id}
+			innerHtml={stripHtmlEmojis( innerHtml, bullets )}
+			tagName={tagName}
+		/>
+	</Fragment>
+);
+
+EmojiList.propTypes = {
+	bullets: PropTypes.arrayOf( PropTypes.string ).isRequired,
+	id: PropTypes.string.isRequired,
+	innerHtml: PropTypes.string.isRequired,
+	tagName: PropTypes.oneOf( [ 'ol', 'ul' ] ).isRequired,
+};
+
+EmojiList.defaultProps = {
+	bullets: [],
+	tagName: 'ul',
+};
+
+/**
+ * An emoji list of arbitrary content.
+ */
+export const StructuredEmojiList = ( {
+	items,
+	id,
+} ) => {
+	const listStyle = items
+		.map( ( item, i ) => getBulletStyle( item.bullet, id, i ) )
+		.join( '' );
+
+	return (
+		<ul className={cx( 'container', id )}>
+			<style dangerouslySetInnerHTML={{ __html: listStyle }}/>
+			{items.map( ( { item }, i ) => (
+				<li key={i} className={cx( id )}>
+					{item}
+				</li>
+			) )}
+		</ul>
+	);
+};
+
+StructuredEmojiList.defaultProps = {
+	items: [],
+};
+
+StructuredEmojiList.propTypes = {
+	id: PropTypes.string.isRequired,
+	items: PropTypes.arrayOf( PropTypes.shape( {
+		bullet: PropTypes.string,
+		item: PropTypes.node,
+	} ) ).isRequired,
+};
+
+export default EmojiList;
```

Comment:
If the end result is that we export both as siblings, i'd rather we export them as siblings here, too